src/osd/PGLog.cc

```diff
@@ -122,7 +122,7 @@ void PGLog::IndexedLog::trim(
 
   while (!dups.empty()) {
     const auto& e = *dups.begin();
-    if (e.version.version >= earliest_dup_version)
+    if (e.version.version > earliest_dup_version)
```

Comment:
could you add a unit test for this? unittest_pglog (test/osd/TestPGLog.cc) has some for trim already